ethereum/contracts/query/QueryResponse.sol

```diff
@@ -73,18 +74,33 @@ error RequestTypeMismatch();
 error UnsupportedQueryType();
 error UnexpectedNumberOfResults();
 error InvalidPayloadLength(uint256 received, uint256 expected);
+error InvalidContractAddress();
+error InvalidFunctionSignature();
+error InvalidChainId();
+error InvalidBlockNum();
+error InvalidBlockTime();
 
 // @dev QueryResponse is a library that implements the parsing and verification of Cross Chain Query (CCQ) responses.
 abstract contract QueryResponse {
     using BytesParsing for bytes;
 
+    IWormhole public immutable wormhole;
```

Comment:
I don't think we should have state defined in a helper contract unless it is stored at a deterministic address. @kcsongor for best practice recommendations here.

Reply:
It's immutable so it's not really state (it's inlined into the bytecode at compile time), plus if you have to inherit the helper contract anyway (because it's abstract) I'm not sure why declaring this in the helper is a problem? But intrigued to hear the argument against this

Reply:
Ah, fair. Def right considering the immutable state. That said, I'm wondering if it _should_ be immutable - it's (very remotely) possible that an integrator may want / need to switch this.

Reply:
Sounds like @kcsongor is in favor of immutable use here. So I'll close this.